docs/locale.mdx

```diff
@@ -0,0 +1,181 @@
+`plugins/Essentials/messages/` folder. In this directory, create an empty file named `messages_<lang>.properties`,
+where `<lang>` is the language code for the language you want to edit. You can now insert any customised messages in
+this file, and they will be loaded by EssentialsX the next time you restart your server. For example:
+
+```properties
+backOther=<primary>Returned <secondary>{0}<primary> to previous location.
+```
+
+:::note
+The numbers in curly braces (ie `{0}`, `{1}`) are template placeholders, which EssentialsX will replace as appropriate
+before displaying the message. You can compare the messages in the default message files with the appearance in-game to
+determine which messages are needed.
+:::
+
+The colors defined in the `config.yml` file can also be used in your custom messages, and will be applied to text inside
+`<primary>Gold text here</primary>` and `<secondary>Red text here</secondary>` tags.
+
+:::tip
+You can search through the default messages files on
+[GitHub](https://github.com/EssentialsX/Essentials/tree/2.x/Essentials/src/main/resources). We strongly recommend
+copying only the messages you want to edit, as this will make it easier to manage and update these files across
+EssentialsX versions.
+:::
+
+:::warning Do not edit any messages files directly inside the EssentialsX jar
+You will lose these changes when you update the plugin and they will not be converted to the new MiniMessage format
+automatically.
+:::
+
+### Using rich text elements
+
+**NEW IN 2.21.0:** EssentialsX now supports the MiniMessage syntax for its messages. This allows you to include
+gradients, hoverable and clickable elements, custom fonts and more!
+
+Below is an example of hoverable/clickable elements in an EssentialsX message:
+
+```properties
+backAfterDeath=<primary><hover:show_text:'Click to go back'><click:run_command:'/back'><secondary>Click here</secondary>to return to your death point.
+```
+
+You can find a full reference guide for the MiniMessage syntax on the
+[Adventure wiki](https://docs.advntr.dev/minimessage/format.html). You can also use Adventure's
+[MiniMessageViewer](https://webui.advntr.dev/) to preview custom messages (although this doesn't support EssentialsX's
+`<primary>` and `<secondary>` tags).
+
+### Hiding specific messages
+
+Most messages can be disabled or hidden - simply set the message to blank. For example:
+
+```properties
+noNewMail=
+```
+
+### Editing messages in the config file
+
+There are a small number of settings in EssentialsX which can be altered by editing `plugins/Essentials/config.yml`,
+such as the currency format, and custom join and leave messages.
+
+### Creating custom text commands
+
+<!-- Text Commands link to be updated. -->
+EssentialsX's text commands can be edited through text files in the `plugins/Essentials` directory. For more
+information, see [Text Commands](https://essentialsx.net/wiki/Text-Commands.html).
```

Comment:
this will need to be added later on, will add to the [to-do list](https://github.com/users/Evidentsinger14/projects/1/views/1&itemId=63201966)
mark as outdated for the sake of the review.